<launch>
    <!-- Launch the base.launch file -->
    <include file="$(find oradar_lidar)/launch/laser_scan.launch"/>
    <!-- 启动激光雷达避障节点 -->
    <!-- Activate lidar obstacle avoidance node -->
    <node name='laser_scan_filter_1' pkg="laser_test" type="laser_scan_filter.py" required="true" output="screen">

        <param name="detect_angle_min" type="int"   value="-45"/>		<!--Minimum Detect Angle-->
        <param name="detect_angle_max" type="int"   value="135"/>		<!--Maximum Detect Angle-->
        <param name="ResponseDist_max" type="double"   value="1.5"/>	<!--Maximum Detect Distance-->
        <param name="ResponseDist_min" type="double"   value="0.15"/>	<!--Minimum Detect Distance-->
        <param name="topic_subscribed" value="/scan1"/>				    <!--ROS Topic Subscribed-->
        <param name="topic_published"  value="/filtered1"/>				<!--ROS Topic Published-->
        <param name="frame_id"  value="/laser1"/>			            <!--Frame ID-->
    
    </node>

    <node name='laser_scan_filter_2' pkg="laser_test" type="laser_scan_filter.py" required="true" output="screen">

        <param name="detect_angle_min" type="int"   value="-135"/>		<!--Minimum Detect Angle-->
        <param name="detect_angle_max" type="int"   value="45"/>		<!--Maximum Detect Angle-->
        <param name="ResponseDist_max" type="double"   value="1.5"/>	<!--Maximum Detect Distance-->
        <param name="ResponseDist_min" type="double"   value="0.15"/>	<!--Minimum Detect Distance-->
        <param name="topic_subscribed" value="/scan2"/>				    <!--ROS Topic Subscribed-->
        <param name="topic_published" value="/filtered2"/>				<!--ROS Topic Published-->
        <param name="frame_id"  value="/laser2"/>			            <!--Frame ID-->
    
    </node>

    <node name='laser_scan_filter_3' pkg="laser_test" type="laser_scan_filter.py" required="true" output="screen">

        <param name="detect_angle_min" type="int"   value="90"/>		<!--Minimum Detect Angle-->
        <param name="detect_angle_max" type="int"   value="270"/>		<!--Maximum Detect Angle-->
        <param name="coordinate_360_degree"  value="True"/>			            <!--True if using 360 Coordinate, otherwise use 0~180 & 0~-180 coordinate-->
        <param name="ResponseDist_max" type="double"   value="1.5"/>	<!--Maximum Detect Distance-->
        <param name="ResponseDist_min" type="double"   value="0.15"/>	<!--Minimum Detect Distance-->
        <param name="topic_subscribed" value="/scan3"/>				    <!--ROS Topic Subscribed-->
        <param name="topic_published" value="/filtered3"/>				<!--ROS Topic Published-->
        <param name="frame_id"  value="/laser3"/>			            <!--Frame ID-->
            
    </node>

</launch>
